### PR TITLE
guide: expand Experiments guide

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Restructures the guide into an intro and a few pages.

Per #2548 (main files changed linked in each item below)

- [x] Copy edit https://github.com/iterative/dvc.org/pull/2618 ([file changes](https://github.com/iterative/dvc.org/pull/2654/files?authenticity_token=k53ZEbxYcI77Rntmd%2F88fdbqSzBJQG9kq41%2B3sW2npFBko0txa9AdJPbx%2Fgf9W6FzJhWIlvCitKufyRYsGKPTg%3D%3D&file-filters%5B%5D=.json#diff-1b150c58a1c943d3149b36bf01defdcac47b31b74eed095adad8298827c3d9c4)) via https://github.com/iterative/dvc.org/pull/2711 and https://github.com/iterative/dvc.org/pull/2719 (nested into this PR)
- [x] ~~Review~~ Copy edit [Checkpoints guide](https://github.com/iterative/dvc.org/pull/2654/files#diff-b9806bbdaf8e3ac58a0a85b160ac6853229c83336abf60e24e2556c3ae054186) for now. ~~Can it be broader yet shorter?  
  Move deep details from https://dvc.org/doc/command-reference/exp/run#checkpoints
  Done in https://github.com/iterative/dvc.org/pull/2753 (not nested)~~
  Extracted to https://github.com/iterative/dvc.org/issues/2769
- [x] Extracted **DVC Experiments** section [from index](https://github.com/iterative/dvc.org/pull/2654/files?authenticity_token=k53ZEbxYcI77Rntmd%2F88fdbqSzBJQG9kq41%2B3sW2npFBko0txa9AdJPbx%2Fgf9W6FzJhWIlvCitKufyRYsGKPTg%3D%3D&file-filters%5B%5D=.json#diff-128fef4265ae36168840329951fb19fd9ce86c7fa965fc244bd024f3ec96346e) into [it's own page](https://github.com/iterative/dvc.org/pull/2654/files?authenticity_token=k53ZEbxYcI77Rntmd%2F88fdbqSzBJQG9kq41%2B3sW2npFBko0txa9AdJPbx%2Fgf9W6FzJhWIlvCitKufyRYsGKPTg%3D%3D&file-filters%5B%5D=.json#diff-17f833349dd6c62cd54dfedc4c73234be01aab006300e4ef9c2879fc1b1d4e72).
- [x] Related links and some other copy edits

Potentially:

- [ ] Move any remaining details and explanations about `dvc exp` from the cmd ref.
- [ ] Remove pipeline concept from dvc-experiments and running-experiments (cross-listed in #2768 and #2883)
- [ ] https://github.com/iterative/dvc.org/issues/2687

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
- [ ] Add info from https://github.com/iterative/dvc.org/pull/2672 to the base checkpoints page
  And/or merge base checkpoints and full guide?

**Comment by jorgeorpinel** [comment]
@shcheklein @iesahin this is the first in a series of cascading or nested PRs to restructure the Experiments guide and move most of the details from the command ref (for now only restructuring). Should I keep it as a draft until more PRs are open against this branch? Should we start the review process for this part? (It's ready) Thanks

File: content/docs/user-guide/experiment-management/experiments.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+## Experiments
+
+_New in DVC 2.0_
+
+`dvc exp` commands let you automatically track a variation to an established
```
Comment (shcheklein) [inline content/docs/user-guide/experiment-management/experiments.md:5]: not for this PR: `track a variation to an established data pipeline` sounds very complicated. 

`dvc exp run`, `dvc exp show`, and other `dvc exp` commands automatically capture and save experiment runs, including code, data, metrics, models, etc. 

or even better - get rid of this sentence :) It does more harm than good to my mind.

File: content/docs/user-guide/experiment-management/experiments.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+[data pipeline](/doc/command-reference/dag). You can create multiple isolated
+experiments this way, as well as review, compare, and restore them later, or
+roll back to the baseline. The basic workflow goes like this:
+
+- Modify stage <abbr>parameters</abbr> or other dependencies (e.g. input data,
```
Comment (shcheklein) [inline content/docs/user-guide/experiment-management/experiments.md:10]: not for this PR: Modify hyperparameters (link to params), code ... etc ... we are being way too formal here again. I thinks we can sacrifice it a bit and use some common ML terminology

**Comment by shcheklein** [comment]
@jorgeorpinel are there any changes besides the split itself? (to help us review it)

File: content/docs/user-guide/experiment-management/checkpoints.md [hunk 3]
```diff
@@ -1,27 +1,32 @@
...
-want to track, which you'll see later in this tutorial.
+Like with regular experiments, checkpoints can become persistent by
+[committing them to Git](#committing-checkpoints-to-git).
 
-This tutorial is going to cover how to implement checkpoints in an ML project
-using DVC. We're going to train a model to identify handwritten digits based on
-the MNIST dataset.
+This guide covers how to implement checkpoints in an ML project using DVC. We're
+going to train a model to identify handwritten digits based on the MNIST
+dataset.
```
Comment (shcheklein) [inline content/docs/user-guide/experiment-management/checkpoints.md:29]: Not this PR: below: Setting up the project ... should we do `##` -> `###`?

Comment (shcheklein) [reply content/docs/user-guide/experiment-management/checkpoints.md:29]: everywhere `bash` -> `dvc`

File: content/docs/sidebar.json [hunk 4]
```diff
@@ -139,7 +139,7 @@
         "label": "Experiment Management",
         "slug": "experiment-management",
         "source": "experiment-management/index.md",
-        "children": ["checkpoints"]
+        "children": ["experiments", "checkpoints", "organization"]
```
Comment (shcheklein) [inline content/docs/sidebar.json:142]: organization -> Organizing Project ?
organization is way to abstract ... essentially it's about how do you structure the workflow? 

to be honest very hard to tell if the existing structure makes sense - would be great to see more content that you have in mind for each section

**Comment by jorgeorpinel** [comment]
> are there any changes besides the split itself?

Minor rewriting of the previous content. Hard to see in the diffs since a lot got moved around but nothing major just better explanations and terminology.

**Comment by shcheklein** [comment]
Okay, I would suggest to move forward with this (or even merge?) Right now, the value of having different section is not exactly clear, names and structure also :) We need a bit more to start reviewing it. Or some plan on how do you want to proceed with this.

File: content/docs/sidebar.json (hunk 4 above)
Comment (jorgeorpinel) [reply content/docs/sidebar.json]: I merged the **Organization Patterns** section back into the index so it has more context. If we get more content for this we can split and rename.

**Comment by jorgeorpinel** [comment]
> I would suggest to move forward with this (or even merge?)

If you guys approve then we can merge this, sure. Could make sense since #2618 has been merged already (already combined with this branch to).

See 

> some plan on how do you want to proceed with this.

This was meant as a first step towards adding Exp Mgmt guide content. The PRs that would follow (nested against this one) would be based on the TODOs from https://github.com/iterative/dvc.org/issues/2548. In fact I added some checkboxes to the description in this PR about what exactly will follow.

**Comment by jorgeorpinel** [comment]
@iesahin @shcheklein WDYT about the changes here? Please see the description for a list (and links to the related file changes). Especially about having a dedicated **DVC Experiments** page to explain what experiments are and potentially go into implementation details (in the future). It would make more sense with a simplification of the **Running Experiments** page but let's leave that for another PR (it's a task in #2768 for now).

See https://dvc-org-guide-exps-i6pyd2tmqgh.herokuapp.com/doc/user-guide/experiment-management

p.s. Looks like nesting PRs wasn't very successful, this ended up being quite a big PR after merging a few others back into it... Sorry!

File: content/docs/user-guide/experiment-management/index.md [hunk 5]
```diff
@@ -1,89 +1,73 @@
...
-- Use [metrics](/doc/command-reference/metrics) to identify the best
-  experiment(s).
-- Visualize, compare experiments with `dvc exp show` or `dvc exp diff`. Repeat
-  🔄
-- Use `dvc exp apply` to roll back to the best one.
-- Make the selected experiment persistent by committing its results to Git. This
-  cleans the slate so you can repeat the process.
-
-## Checkpoints in source code
+Every time you [reproduce](/doc/command-reference/repro) a pipeline with DVC, it
```
Comment (shcheklein) [inline content/docs/user-guide/experiment-management/index.md:19]: sounds too much for the index page? and too abrupt to be honest (even though it is in the details section)

File: content/docs/user-guide/experiment-management/index.md [hunk 6]
```diff
@@ -1,89 +1,73 @@
...
-`dvc exp` workflow (refer to previous sections), the `dvc.yaml` and `dvc.lock`
-pair in the <abbr>workspace</abbr> will codify the experiment as a new project
-version. The right <abbr>outputs</abbr> (including
-[metrics](/doc/command-reference/metrics)) should also be present, or available
-via `dvc checkout`.
+> 👨‍💻 See [Get Started: Experiments](/doc/start/experiments) for a hands-on
+> introduction to DVC experiments.
 
-### Organization patterns
+### Organization Patterns
```
Comment (shcheklein) [inline content/docs/user-guide/experiment-management/index.md:67]: should it be somewhere inside? (not on the index page)?

**Comment by shcheklein** [comment]
@jorgeorpinel there are indeed too many changes at once, some of them are not that related. Could you please split this? otherwise we'll have a few reviews happening at once in one PR - it can take forever to do this :)

File: content/docs/user-guide/experiment-management/dvc-experiments.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+## DVC Experiments
```
[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply content/docs/user-guide/experiment-management/dvc-experiments.md:1]: p.p.s. that I said I'd understand if you prefer me to split this into several PRs (one per check box above roughly). Can do. Lmk

**Comment by jorgeorpinel** [comment]
OK. I'll split this ⌛

**Comment by jorgeorpinel** [comment]
> - [x] Extracted **DVC Experiments** section [from index](https://github.com/iterative/dvc.org/pull/2654/files?authenticity_token=k53ZEbxYcI77Rntmd%2F88fdbqSzBJQG9kq41%2B3sW2npFBko0txa9AdJPbx%2Fgf9W6FzJhWIlvCitKufyRYsGKPTg%3D%3D&file-filters%5B%5D=.json#diff-128fef4265ae36168840329951fb19fd9ce86c7fa965fc244bd024f3ec96346e) into [it's own page](https://github.com/iterative/dvc.org/pull/2654/files?authenticity_token=k53ZEbxYcI77Rntmd%2F88fdbqSzBJQG9kq41%2B3sW2npFBko0txa9AdJPbx%2Fgf9W6FzJhWIlvCitKufyRYsGKPTg%3D%3D&file-filters%5B%5D=.json#diff-17f833349dd6c62cd54dfedc4c73234be01aab006300e4ef9c2879fc1b1d4e72).
> - [x] Related links and some other copy edits

Extracted to https://github.com/iterative/dvc.org/pull/2909

**Comment by jorgeorpinel** [comment]
> - [x] Copy edit [Checkpoints guide](https://github.com/iterative/dvc.org/pull/2654/files#diff-b9806bbdaf8e3ac58a0a85b160ac6853229c83336abf60e24e2556c3ae054186) for now. ~~Can it be broader yet shorter?  

Extracted to https://github.com/iterative/dvc.org/pull/2910

**Pull request closed without merging**